core/storage/auth/gae_models.py

```diff
@@ -20,11 +20,226 @@
 from __future__ import unicode_literals  # pylint: disable=import-only-modules
 
 from core.platform import models
+import feconf
 
-base_models, = models.Registry.import_models([models.NAMES.base_model])
+base_models, user_models = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
 datastore_services = models.Registry.import_datastore_services()
 
 
+class UserAuthDetailsModel(base_models.BaseModel):
+    """Stores the authentication details for a particular user.
+
+    Instances of this class are keyed by user id.
+    """
+
+    # Authentication detail for sign-in using google id (GAE). Exists only
+    # for full users. None for profile users.
+    gae_id = datastore_services.StringProperty(indexed=True)
+    # Authentication detail for sign-in using Firebase authentication.
+    # TODO(#11462): Exists for all users.
```

Comment:
TODOs should directive and specify what needs to happen. I don't understand this one...